README.md
[![PyPI](https://img.shields.io/pypi/v/dmri-amico)](https://pypi.org/project/dmri-amico/)
[![PyPI - Downloads](https://img.shields.io/pypi/dm/dmri-amico)](#)
[![GitHub](https://img.shields.io/github/license/daducci/AMICO)](#)
[![GitHub top language](https://img.shields.io/github/languages/top/daducci/amico?color=lightgray)](#)
[![reference](https://img.shields.io/badge/DOI-10.1016/j.neuroimage.2014.10.026-red.svg)](https://www.sciencedirect.com/science/article/pii/S1053811914008519)

[![GitHub stars](https://img.shields.io/github/stars/daducci/AMICO?style=social)](#)
[![GitHub forks](https://img.shields.io/github/forks/daducci/AMICO?style=social)](#)
[![GitHub watchers](https://img.shields.io/github/watchers/daducci/AMICO?style=social)](#)
[![GitHub followers](https://img.shields.io/github/followers/daducci?style=social)](#)
[![GitHub contributors](https://img.shields.io/github/contributors-anon/daducci/AMICO?style=social&link)](#)
[![Twitter Follow](https://img.shields.io/twitter/follow/ADaducci)](https://twitter.com/intent/follow?screen_name=ADaducci)

![example workflow](https://github.com/nightwnvol/amico_gh_actions/actions/workflows/compile_openblas_macos_x86_64.yml/badge.svg)
![example workflow](https://github.com/nightwnvol/amico_gh_actions/actions/workflows/compile_openblas_linux.yml/badge.svg)
![example workflow](https://github.com/nightwnvol/amico_gh_actions/actions/workflows/build_wheels.yml/badge.svg)
![example workflow](https://github.com/nightwnvol/amico_gh_actions/actions/workflows/macos_arm64_wheels.yml/badge.svg)

![Build Status](https://api.cirrus-ci.com/github/nightwnvol/amico_gh_actions.svg?task=Compile%20OpenBLAS%20on%20MacOS%20arm64)
![Build Status](https://api.cirrus-ci.com/github/nightwnvol/amico_gh_actions.svg?task=Build%20MacOS%20arm64%20wheels)


# AMICO

Implementation of the linear framework for Accelerated Microstructure Imaging via Convex Optimization (AMICO) described here:

> **Accelerated Microstructure Imaging via Convex Optimization (AMICO) from diffusion MRI data**  
> *Alessandro Daducci, Erick Canales-Rodriguez, Hui Zhang, Tim Dyrby, Daniel Alexander, Jean-Philippe Thiran*  
> NeuroImage 105, pp. 32-44 (2015)

## Code implementation

AMICO is written in **python** and **does not require** the [NODDI MATLAB toolbox](http://mig.cs.ucl.ac.uk/index.php?n=Download.NODDI) to be present on your system; all the necessary MATLAB functions for generating the response functions of the NODDI model have in fact been ported to Python.

# Documentation

The documentation, including the [installation guide](https://github.com/daducci/AMICO/wiki/Installation) as well as a series of demos/tutorials to show how to use AMICO to fit different models to the data, can be found in the [wiki pages](https://github.com/daducci/AMICO/wiki/Home).

For questions, requesting assistance, suggesting enhancements or new ideas as well as for reporting bugs, please open an [issue](https://github.com/daducci/AMICO/issues).


CHANGELOG.md
# Change Log
All notable changes to AMICO will be documented in this file.

## [2.0.0] - 2022-##-##

### Changed
- Implement 'fit' method of the models in Cython (models.pyx)
- Implement the 'dir_to_lut_idx' method in Cython (lut.pyx)
- Implement multithreading

### Added
- C++ solvers (NNLS and LASSO)
- 'DTI_fit_method' config to choose the fitting method for the Diffusion Tensor model (dipy)
- 'BLAS_nthreads' config to set the number of threads used in the threadpool-backend of common BLAS implementations (e.g. OpenBLAS)
- Loader-like context manager 'util.Loader()'

## [1.5.1] - 2022-09-21

### Fixed
- Check if DWI file is a 4D image
- Removed unused 'verbose' option in __init__ (now all is controleld by 'set_verbose')

## [1.5.0] - 2022-07-06

### Changed
- Implemented the 'synthesis.py' module for kernels generation
- 'datasynth' tool (CAMINO) no more needed

## [1.4.4] - 2022-06-14

### Fixed
- Import error in 'util.py' on Windows systems
- Loading errors in 'lut.py' on Windows systems

## [1.4.3] - 2022-05-18

### Fixed
- Problems with joblib and pickle on some systems (issue #136)
- Added 'packaging' to requirements as it is not present by default on some systems

### Changed
- Removed config option 'parallel_backend' as only 'loky' works in current implementation

## [1.4.2] - 2022-03-26

### Fixed
- Bug when using 'doSaveCorrectedDWI' (issue #134)

## [1.4.1] - 2022-03-20

### Fixed
- Replace all 'print' with 'PRINT'

## [1.4.0] - 2022-03-04

### Added
- Function 'amico.set_verbose' to control what is printed during execution

### Changed
- Required dependency from 'python-spams' to 'spams'

## [1.3.2] - 2022-01-31

### Changed
- Required dependency numpy>=1.12,<1.22
- amico.core.setup: removed 'ndirs' parameter, now precomputes all directions

### Added
- Shortcut 'amico.core.setup' to 'amico.setup'

## [1.3.1] - 2021-12-03

### Fixed
- Removed unused hasISO parameter in experimental VolumeFractions model

### Changed
- Install information are stored (and taken from) amico/info.py

## [1.3.0] - 2021-10-27

### Added
- Possibility to fit different voxels in parallel
- Config options 'parallel_jobs' and 'parallel_backend' for better control of parallel execution
- Possibility to specify d_perp<>0  for the Stick in StickZeppelinBall

### Fixed
- Forcing SPAMS to use only one thread
- Added missing dependencies, e.g. wheel

### Changed
- Replaced in-house progress bar to tqdm

## [1.2.10] - 2021-05-20

### Fixed
- Warning message in util.py fsl2scheme and sandi2scheme

## [1.2.9] - 2021-02-23

### Added
- Controls if files exist before opening

### Fixed
- Wrong datatype when saving results

## [1.2.8] - 2021-01-29

### Added
- SANDI model

## [1.2.7] - 2020-10-28

### Added
- Possibility to set one single direction in the LUT resolution

## [1.2.6] - 2020-10-22

### Added
- Function get_params to get the specific parameter values of a model

## [1.2.5] - 2020-08-06

### Changed
- Moved the documentation to the Wiki

## [1.2.4] - 2020-06-10

### Added
- Added the parameter 'd_par_zep' in the StickZeppelinBall model

## [1.2.3] - 2020-05-25

### Fixed
- Modify setup.py and fix spams dependency

## [1.2.2] - 2020-05-05

### Fixed
- Checks if input mask is 3D.

## [1.2.1] - 2020-04-25

### Changed
- Use d_perps to set models instead of ICVFs.
- Canged case of some parameters for consistency

## [1.2.0] - 2020-04-01

### Added
- Functions to colorize the output messages.

## [1.1.0] - 2019-10-30

This version of AMICO *is not compatible* with [COMMIT](https://github.com/daducci/COMMIT) v1.2 of below. If you update AMICO to this version, please update COMMIT to version 1.3 or above.

### Added
- Changelog file to keep tracking of the AMICO versions.
- New feature that allows to decrease the LUT resolution. [Example here.](https://github.com/ErickHernandezGutierrez/AMICO/blob/lowresLUT/doc/demos/NODDI_lowres.md)


amico/info.py
# -*- coding: UTF-8 -*-

# Format version as expected by setup.py (string of form "X.Y.Z")
_version_major = 2
_version_minor = 0
_version_micro = 0
_version_extra = '' #'.dev'
__version__    = "%s.%s.%s%s" % (_version_major,_version_minor,_version_micro,_version_extra)

NAME                = 'dmri-amico'
DESCRIPTION         = 'Accelerated Microstructure Imaging via Convex Optimization (AMICO)'
LONG_DESCRIPTION    = """
=======
 AMICO
=======

Implementation of the linear framework for Accelerated Microstructure Imaging via Convex Optimization (AMICO) described here:

| Accelerated Microstructure Imaging via Convex Optimization (AMICO) from diffusion MRI data
| Alessandro Daducci, Erick Canales-Rodriguez, Hui Zhang, Tim Dyrby, Daniel Alexander, Jean-Philippe Thiran
| NeuroImage 105, pp. 32-44 (2015)
"""
URL                 = 'https://github.com/daducci/AMICO'
DOWNLOAD_URL        = "N/A"
LICENSE             = 'BSD license'
AUTHOR              = 'Alessandro Daducci'
AUTHOR_EMAIL        = 'alessandro.daducci@univr.it'
PLATFORMS           = "OS independent"
MAJOR               = _version_major
MINOR               = _version_minor
MICRO               = _version_micro
VERSION             = __version__
